src/api/private/notes/notes.controller.ts

```diff
@@ -174,6 +178,84 @@ export class NotesController {
     }
   }
 
+  @Post(':noteIdOrAlias/metadata/alias/:alias')
+  async addAlias(
+    @Req() req: Request,
+    @Param('noteIdOrAlias') noteIdOrAlias: string,
+    @Param('alias') alias: string,
+  ): Promise<NoteMetadataDto> {
+    try {
+      // ToDo: use actual user here
+      const user = await this.userService.getUserByUsername('hardcoded');
+      const note = await this.noteService.getNoteByIdOrAlias(noteIdOrAlias);
+      if (!this.permissionsService.isOwner(user, note)) {
+        throw new UnauthorizedException('Reading note denied!');
+      }
+      const updatedNote = await this.noteService.addAlias(note, alias);
+      return await this.noteService.toNoteMetadataDto(updatedNote);
+    } catch (e) {
+      if (e instanceof AlreadyInDBError) {
+        throw new BadRequestException(e.message);
+      }
+      if (e instanceof ForbiddenIdError) {
+        throw new BadRequestException(e.message);
+      }
+      throw e;
+    }
+  }
+
+  @Put(':noteIdOrAlias/metadata/alias/:alias')
+  async makeAliasPrimary(
+    @Req() req: Request,
+    @Param('noteIdOrAlias') noteIdOrAlias: string,
+    @Param('alias') alias: string,
```

Comment:
Fair enough.

Reply:
Or not. This request while possible in theory is a bit problematic:

The path parameter noteIdOrAlias will be there anyway, so removing is only a cosmetic change. But you would also need to put the handling of incorrect aliases into the controller instead of the service, something I'm uncomfortable
with. I think it's best to keep it like it is…

Reply:
What would it change to put `noteIdOrAlias` as parameter of `getNoteByIdOrAlias` and `makeAliasPrimary`? I think incorrect aliases will be found by `getNoteByIdOrAlias`.

Reply:
Yes but this way I would allow semantically problematic requests. Any note identifier including nonexistent and ones for other notes could be include in the url to make an alias primary or delete it altogether. I don't think we want that.

Reply:
I misunderstood what you wanted and I implemented your request changing the primary alias is now done via `PUT :newPrimaryAlias/metadata/alias`

Reply:
Yeah it's maybe also better to rename `:noteIdOrAlias` to `:alias` in the route because it could be confusing because noteIds can't be the primary alias.

Reply:
changed